### DISCUSSION TITLE
Cannot Post /login/callback

### DISCUSSION THREAD

**Discussion by BrendonKupsch1** [bergie/passport-saml#947] [Q&A]

**Comment by markstos** [comment]
Your app not working is not a bug in `passport-saml`. I've moved this to Discussions.

**Comment by srd90** [comment]
>... after a successful login I get routed to my callback URL and get a blank page with the error Cannot POST /login/callback There is no information in the console or SAML tracer other than a 404 error, and logs on the server have not helped either. Any help would be appreciated.
>...
>```javascript
>...
>// SSO callback route
>app.post(
>  'https://facelect.capping.ecrl.marist.edu/login/callback',
>...
>```
>...

Dunno about you but I see smelly code (configuration). I am not an expressjs user but IMHO [your passport-saml bug report](https://github.com/node-saml/passport-saml/issues/946) is actually your express router definition error. Have you tried `app.post('/login/callback', ...`? See also https://expressjs.com/en/guide/routing.html

**Reply by BrendonKupsch1** [reply]
`app.post('/login/callback', ...` was my first solution, I get the same Cannot POST error.

**Reply by srd90** [reply]
You could debug this with any dummy implementation (which returns anything for get/post) because it was/is obvious that this is not passport-saml bug.

Here are few more hints:

I see that you launch your server to port 3001 but you have tested it by performing HTTP GET (curl's default if not specified e.g. with `-XPOST`) to your domain name 443 port and managed to get HTTP response code 200. This would implicate some sort of intercepting reverse proxy between your application and your curl. Check that proxy's configs to see whether POST is allowed.

**Reply by srd90** [reply]
Well, IF you are using stuff that you have posted:

1. Your POST route definition is still incorrect as communicated earlier.
2. You have launched your express service to port 3001 
   >```javascript
   >// Create HTTPS server on port 3001
   >https.createServer(options, app).listen(3001, () => {
   >   console.log('HTTPS Server running on port 3001');
   >});
   >```
   
   but your proxypass definition says 3000...i.e. you are not directing traffic to your SP at 3001...dunno who answers to your GET with 200 and with 404 to POST...take a look who is behind 3000
   >```
   >   ProxyPreserveHost On
   >   ProxyPass / http://localhost:3000/
   >   ProxyPassReverse / http://localhost:3000/
   >```


>If you need to look at any other apache2 files I will provide them.

This was end of discussion at least from my side. I'm not going to provide remote debug support for apache & express traffic routing issues via GH issue/discussion transport bearer.

BTW. If you continue this thread provide exact steps to reproduce with exact configuration you have at the time of writing. Someone might take a look or not.

**Reply by srd90** [reply]
Question was closed by @BrendonKupsch1 without followup comment and content of initial message was deleted along with comment that revealed incorrect reverse proxy configuration.

So for anyone who lands to this discussion e.g. via search engine:

Situation was that this endpoint didn't receive HTTP POST requests:
* https://github.com/KaiCottle/electionCapping24/blob/63c3cf04fdfaf6f0d901e1bc8787686d6d0d6ea7/server.js#L77-L88

and it was sitting behind port 3001:
* https://github.com/KaiCottle/electionCapping24/blob/63c3cf04fdfaf6f0d901e1bc8787686d6d0d6ea7/server.js#L156-L159

Based on timestamps it seems that this discussion was marked closed quite soon after/about same time that some commits were made to move [BE behind port where reverse proxy was directing traffic](https://github.com/node-saml/passport-saml/discussions/947#discussioncomment-11495628):
* https://github.com/KaiCottle/electionCapping24/compare/63c3cf04fdfaf6f0d901e1bc8787686d6d0d6ea7..22ee0535ee9af3a798408e5064ba78e08055391a

Based on aforementioned commits it is quite safe to assume that at the end of the day root cause in this particular case was incorrect reverse proxy definition (traffic was terminated to some random service instead of SP).
